pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/ForwardIndexReader.java

```diff
@@ -916,4 +916,83 @@ default byte[][] getBytesMV(int docId, T context) {
   default int getNumValuesMV(int docId, T context) {
     throw new UnsupportedOperationException();
   }
+
+  class ValueRange {
+    private final long _offset;
+    private final int _size;
+    // To tell if size uses bit or byte as unit, for fwd index reader reading values of fixed bits.
+    private final boolean _isSizeOfBit;
+
+    public static ValueRange newByteRange(long offset, int sizeInBytes) {
+      return new ValueRange(offset, sizeInBytes, false);
+    }
+
+    public static ValueRange newBitRange(long offset, int sizeInBits) {
+      return new ValueRange(offset, sizeInBits, true);
+    }
+
+    private ValueRange(long offset, int size, boolean isSizeOfBit) {
+      _offset = offset;
+      _size = size;
+      _isSizeOfBit = isSizeOfBit;
+    }
+
+    public long getOffset() {
+      return _offset;
+    }
+
+    public int getSize() {
+      return _size;
+    }
+
+    public boolean isSizeOfBit() {
+      return _isSizeOfBit;
+    }
+
+    @Override
+    public String toString() {
+      return "Range{" + "_offset=" + _offset + ", _size=" + _size + ", _isSizeOfBit=" + _isSizeOfBit + '}';
+    }
+  }
+
+  /**
+   * An interface that enables the caller to get byte ranges accessed while reading a given docId.
+   * This can be used to eventually allow prefetching of exact byte ranges during query processing,
+   * if the docIds to be fetched are known beforehand.
+   *
+   * This interface will be implemented by all the forward index readers.
+   */
+  interface ValueRangeProvider<T extends ForwardIndexReaderContext> {
```

Comment:
What is the reason to do not make all `ForwardIndexReader` extend this method (or just add the methods to `ForwardIndexReader`?

Reply:
I wanted to avoid having to implement it for other implementations like mutable forward index etc

Reply:
It makes sense. Consider it resolved

Reply:
I'd also suggest directly adding it into `ForwardIndexReader`. We may add default implementation for the methods, and potentially add `boolean isByteRangeSupported()`